msaad.go

```diff
@@ -523,6 +524,64 @@ func (m *MSAAD) syncRoles(roleGroups *cachedRoleGroups, aadUser *msaadUser, inte
 	groupsChanged := false
 	userHasAnyIMQSPermission := false
 
+	// identify unmapped groups
+	removeIDs, _ := DecodePermit(make([]byte, 0))
+	allowedIDs, _ := DecodePermit(make([]byte, 0))
+
+	if m.parent.MSAAD.Config.Verbose {
+		m.parent.Log.Debugf("MSAAD empty role arrays constructed")
+	}
+
+	// get all mapped group ids
+	for _, internalGroupName := range m.Config.RoleToGroup {
+		if m.parent.MSAAD.Config.Verbose {
+			m.parent.Log.Debugf("MSAAD checking all roles: %v", internalGroupName)
+		}
+
+		internalGroup, ok := roleGroups.nameToGroup[internalGroupName]
+		if !ok {
+			// We've already logged an error about this, so here we just ignore it
+			continue
+		}
+		m.parent.Log.Debugf("MSAAD add allowed ID for %v", internalGroupName)
+		allowedIDs = append(allowedIDs, internalGroup.ID)
```

Comment:
Please excuse my being pedantic about issues of style
```suggestion
		if internalGroup, ok := roleGroups.nameToGroup[internalGroupName]; ok {
		    m.parent.Log.Debugf("MSAAD add allowed ID for %v", internalGroupName)
		    allowedIDs = append(allowedIDs, internalGroup.ID)
		}
```